### ISSUE TITLE
linter: id-length `properties: "never"` does not exempt TypeScript interface / type-literal property signatures

### ISSUE THREAD

**Issue by Pearce-Ropion** [oxc-project/oxc#26309]
### What version of Oxlint are you using?

1.80.0

### What command did you run?

`oxlint test.ts`

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```jsonc
{
  "rules": {
    "id-length": ["error", { "min": 2, "properties": "never" }]
  }
}
```

### What happened?

The `eslint/id-length` rule's `properties` option is documented as controlling "whether to check property names for length." Setting `properties: "never"` correctly exempts object-literal properties (`{ a: 1 }`) and destructured bindings, but **does not** exempt TypeScript interface members or object-type-literal members (`TSPropertySignature`). These are still checked as if `properties` were `"always"`.

**Reproduction (`test.ts`):**
```ts
interface Foo {
  a: string;
}

const obj = { a: 1 };
```

**Actual output:**
```
test.ts:2:3: error eslint(id-length): Identifier name is too short (< 2).
```

**Expected output:**

No `id-length` error on either `a` — with `properties: "never"`, the object-literal property `a` on line 5 is correctly exempted, so the interface member `a` on line 2 should be exempted for the same reason.

**Likely root cause:**

In `crates/oxc_linter/src/rules/eslint/id_length.rs`, `handle_identifier_name`'s `match parent_node.kind() { ... }` only special-cases `ObjectProperty`, `BindingProperty`, `AssignmentTargetPropertyProperty`, and the member-expression variants for the `properties` gate. There is no arm for `TSPropertySignature` (or `TSMethodSignature`/`TSIndexSignature`), so those identifiers fall through to the default arm, which runs the length check unconditionally regardless of the `properties` setting.

**Suggested fix:**

Add a match arm for `AstKind::TSPropertySignature(_)` (and likely `TSMethodSignature`/`TSIndexSignature`, for consistency) that returns early when `self.properties == AlwaysNever::Never`, mirroring the existing `AstKind::ObjectProperty(_)` arm.

---

Generated by Claude, reviewed by Pearce-Ropion.
